x-pack/plugins/lens/public/editor_frame_service/editor_frame/config_panel/chart_switch/chart_switch_selectable.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+import { EuiSelectable, EuiSelectableOption, EuiSelectableProps, IconType } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { css } from '@emotion/react';
+import { ChartOption } from './chart_option';
+
+export type SelectableEntry = EuiSelectableOption<{
+  value: string;
+  description?: string;
+  icon?: IconType;
+}>;
+
+const ITEM_HEIGHT = 52;
+const MAX_ITEMS_COUNT = 6;
+const MAX_LIST_HEIGHT = ITEM_HEIGHT * MAX_ITEMS_COUNT;
+
+function computeListHeight(list: SelectableEntry[]) {
+  if (list.length > MAX_ITEMS_COUNT) {
+    return MAX_LIST_HEIGHT;
+  }
+}
+
+export const ChartSwitchSelectable = ({
+  setSearchTerm,
+  searchTerm,
+  ...props
+}: { setSearchTerm: (val: string) => void; searchTerm: string } & EuiSelectableProps) => {
+  return (
+    <EuiSelectable
+      singleSelection
+      isPreFiltered
+      data-test-subj="lnsChartSwitchList"
+      className="lnsChartSwitch__options"
+      height={computeListHeight(props.options as SelectableEntry[])}
+      searchProps={{
+        compressed: true,
+        autoFocus: true,
+        className: 'lnsChartSwitch__search',
+        'data-test-subj': 'lnsChartSwitchSearch',
+        onChange: setSearchTerm,
+        placeholder: i18n.translate('xpack.lens.chartSwitch.search', {
+          defaultMessage: 'Search visualizations',
+        }),
+      }}
+      listProps={{
+        showIcons: false,
+        onFocusBadge: false,
+        isVirtualized: false,
+        paddingSize: 'none',
+      }}
+      renderOption={(option, searchValue) => (
+        <ChartOption option={option} searchValue={searchValue} />
+      )}
+      noMatchesMessage={
+        <div
+          css={css`
+            display: inline;
+          `}
+        >
+          <FormattedMessage
+            id="xpack.lens.chartSwitch.noResults"
+            defaultMessage="No results found for {term}."
+            values={{
+              term: <strong>{searchTerm}</strong>,
+            }}
+          />
+        </div>
+      }
+      {...props}
+      searchable
+    >
+      {(list, search) => (
+        <>
+          {search}
```

Comment:
Wrap `search` in an `EuiPopoverTitle` to give it the padding it needs.

```suggestion
          <EuiPopoverTitle paddingSize="s">{search}</EuiPopoverTitle>
```